File: features/document/src/main/java/com/nivasafinance/features/document/storage/ContentRepository.java
package com.nivasafinance.features.document.storage;

import java.io.InputStream;

public interface ContentRepository {
    String saveFile(InputStream inputStream, String documentPath);
    void deleteFile(String storageKey);
    InputStream fetchFile(String storageKey);
    String getSignedDownloadUrl(String storageKey, Long expiresIn);
}

File: features/document/src/main/java/com/nivasafinance/features/document/storage/ContentRepository.kt
package com.nivasafinance.features.document.storage

import java.io.InputStream

interface ContentRepository {
    fun saveFile(inputStream: InputStream, documentPath: String): String
    fun deleteFile(storageKey: String)
    fun fetchFile(storageKey: String): InputStream
    fun getSignedDownloadUrl(storageKey: String, expiresIn: Long): String?
}
